demos/audioplayback/src/audioplaybackapp.cpp

```diff
@@ -91,38 +94,38 @@ namespace nap
 		}
 
 		// Handle playback
-		auto play_back_component = mAudioEntity->findComponent<audio::PlaybackComponentInstance>();
-        if (!play_back_component->isPlaying())
-        {
-            if (ImGui::Button("Play"))
-                play_back_component->start(mStartPosition * 1000.0f, mDuration * 1000.0f);
+		auto pc = mAudioEntity->findComponent<audio::PlaybackComponentInstance>();
+		if (!pc->isPlaying())
+		{
+			if (ImGui::Button("Play"))
+				pc->isFinished() ? pc->start(0.0) : pc->start();
         }
         else
 		{
-            if (ImGui::Button("Stop"))
-                play_back_component->stop();
+			if (ImGui::Button("Stop"))
+				pc->stop();
         }
 
+		// Waveform
+		ImGui::PlotLines("Wave", &mWaveform[0], mWaveform.size(), 0, NULL,
+			mWaveBounds.x, mWaveBounds.y, { 0, ImGui::GetFrameHeight() * 2 });
+
 		// Playback settings
-		float length_seconds = mBuffer->getSize() / (mBuffer->getSampleRate() / 1000.0f) / 1000.0f;
-		ImGui::SliderFloat("Start Position (s)", &mStartPosition, 0, length_seconds, "%.3f", 2);
-		ImGui::SliderFloat("Duration (0 = until end)", &mDuration, 0, 10.0f, "%.3f", 2);
-		if (ImGui::SliderFloat("Fade In (s)", &mFadeInTime, 0, 2.0f, "%.3f", 2))
-		{
-			play_back_component->setFadeInTime(mFadeInTime * 1000.0f);
-		}
-		if (ImGui::SliderFloat("Fade Out (s)", &mFadeOutTime, 0, 2.0f, "%.3f", 2))
-		{
-			play_back_component->setFadeOutTime(mFadeOutTime * 1000.0f);
-		}
-		if (ImGui::SliderFloat("Pitch", &mPitch, 0.5, 2, "%.3f", 1))
-		{
-			play_back_component->setPitch(mPitch);
-		}
-		if (ImGui::SliderFloat("Panning", &mPanning, 0.f, 1.f, "%.3f", 1))
-		{
-			play_back_component->setStereoPanning(mPanning);
-		}
+		float ps = pc->getPosition();
+		if (ImGui::SliderFloat("Position (s)", &ps, 0, pc->getLength(), "%.2f"))
```

Comment:
It's a pity of the clicks and pops when changing the position while playing. I changed to this and it creates a nice "scrubbing" effect. The 5.f is a weird magical number, I tried to derive it from the deltaTime but it didn't work. You have to declare mPitch as a private double member var. 

		// Playback settings
		double oldPos = pc->getPosition();
		float newPos = oldPos;
		bool scrubbing = false;
		if (ImGui::SliderFloat("Position (s)", &newPos, 0, pc->getLength(), "%.2f"))
		{
			if (pc->isPlaying())
			{
				auto pitch = (newPos - oldPos) * 5.f;
				pc->setPitch(pitch);
				scrubbing = true;
			}
			else
				pc->setPosition(newPos);
		}

		ImGui::SliderFloat("Pitch", &mPitch, 0.5, 2, "%.2f");
		if (!scrubbing)
			pc->setPitch(mPitch);